frontend_lib/src/component/EmojiReactions/EmojiReactionButtons.jsx

```diff
@@ -0,0 +1,252 @@
+        5: props.t('Medium-Dark Skin Tone'),
+        6: props.t('Dark Skin Tone')
+      }
+    }
+
+    return (
+      <>
+        <button
+          title={props.t('React')}
+          ref={this.buttonRef}
+          className={classnames('ReactionButton__buttonpicker', { active: this.state.pickerDisplayed })}
+          onClick={() => this.setState(
+            prev => ({
+              pickerDisplayed: !prev.pickerDisplayed
+            })
+          )}
+        >
+          <i className='far fa-smile' />
+        </button>
+        {(this.state.pickerDisplayed
+          ? (
+            <UncontrolledPopover
+              className="EmojiPickerPopover"
+              toggle={props.onCancel} // eslint-disable-line react/jsx-handler-names
+              placement='left'
+              isOpen
+              target={() => this.buttonRef.current}
+            >
+              <PopoverBody>
+                <Picker
+                  native
+                  emojiTooltip={true}
+                  title=''
+                  i18n={emojiMartI18n}
+                  onSelect={this.handleSelect}
+                />
+              </PopoverBody>
+            </UncontrolledPopover>
+          )
+          : undefined
+        )}
+      </>
+    )
+  }
+}
+
+const TranslatedEmojiPickerButton = translate()(EmojiPickerButton)
+
+function getReactionButtonTitle (props) {
+  const reactionList = props.reactionList
+
+  const you = (
+    reactionList[0].reaction_id === props.userReactionId
+      ? props.t('You')
+      : props.t('you')
+  )
+
+  const value = props.reactionList[0].value
+  const emoji = getEmojiDataFromNative(value, 'apple', emojiData)
+
+  const users = andList(
+    props.reactionList.map(
+      ({ author, reaction_id: reactionId }) => (
+        reactionId === props.userReactionId
+          ? you
+          : (author.public_name || author.username)
+      )
+    )
+  )
+
+  const l10nOpts = {
+    reaction: emoji ? (emoji.id || emoji.name) : value,
+    users: users,
+    user: users
+  }
+
+  return (
+    props.userReactionId === -1
+      ? (
+        props.reactionList.length === 1
+          ? props.t('{{user}} reacted with {{reaction}}', l10nOpts)
+          : props.t('{{users}} reacted with {{reaction}}_them', l10nOpts)
+      )
+      : (
+        props.reactionList.length === 1
+          ? props.t('You reacted with {{reaction}}', l10nOpts)
+          : props.t('{{users}} reacted with {{reaction}}_you', l10nOpts)
+      )
+  )
+}
+
+const ReactionButton = translate()(function ReactionButton (props) {
+  const highlighted = props.userReactionId !== -1
+  const title = getReactionButtonTitle(props)
+
+  return (
+    <button
+      onClick={highlighted ? props.onRemoveReaction : props.onAddReaction}
+      title={title}
+      className={classnames('ReactionButton__button', { highlighted })}
+    >
+      <span className='ReactionButton__button__value'>{props.reactionList[0].value}</span>
```

Comment:
You forgot `reactionList` in the `propTypes`.

Reply:
done